packages/components/src/menu-items-choice/types.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import type { ShortcutProps } from '../shortcut/types';
+
+export type MenuItemsChoiceProps = {
+	/**
+	 * Array of choices
+	 */
+	choices: readonly MenuItemChoice[];
```

Comment:
To be honest this is a [copy paste](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__components/menu-items-choice/index.d.ts#L10). I'll let you decide whether that's something we want to keep.

I've definitely seen code which reassigns and/or mutates props but it's not exactly considered a good practice and is usually avoided. As you mentioned I'd consider `readonly` an additional layer of protection which warns you when you're trying to e.g. `.push` something to the `choices` array instead of creating a new array and spread the existing array.

Reply:
I mean, it certainly doesn't hurt! I looked into it and some people actually suggest doing `readonly` by default on all  arrays 😆 I will keep it in mind as well.